File: competition-serv-service/src/main/java/compman/compsrv/service/schedule/internal/InternalFightStartTime.kt
package compman.compsrv.service.schedule.internal

import java.time.Instant

data class InternalFightStartTime(val fightId: String,
                                  val categoryId: String,
                                  val matId: String,
                                  val fightNumber: Int,
                                  val startTime: Instant,
                                  val scheduleEntryId: String,
                                  val periodId: String) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as InternalFightStartTime

        if (fightId != other.fightId) return false

        return true
    }

    override fun hashCode(): Int {
        return fightId.hashCode()
    }
}

File: command-processor/src/main/scala/compman/compsrv/logic/schedule/InternalFightStartTime.scala
package compman.compsrv.logic.schedule

import java.time.Instant

case class InternalFightStartTime(
    fightId: String,
    categoryId: String,
    matId: String,
    fightNumber: Int,
    startTime: Instant,
    scheduleEntryId: String,
    periodId: String
  ) {
    override def equals(obj: Any): Boolean = {
      obj match {
        case time: InternalFightStartTime => fightId.equals(time.fightId)
        case _                            => false
      }
    }

    override def hashCode(): Int = fightId.hashCode
  }